### ISSUE TITLE
Deleting related records

### ISSUE THREAD

**Issue by iworker** [yii2tech/ar-linkmany#4]
When I tried to use delete method for the model with defined behavior, it's relations didn't delete. Maybe it could be useful to add EVENT_AFTER_DELETE handler and unlink all related records.

**Comment by iworker** [comment]
Something like that:

```
    /**
     * @inheritdoc
     */
    public function events()
    {
        return [
            BaseActiveRecord::EVENT_AFTER_INSERT => 'afterSave',
            BaseActiveRecord::EVENT_AFTER_UPDATE => 'afterSave',
            BaseActiveRecord::EVENT_AFTER_DELETE => 'afterDelete',
        ];
    }

    /**
     * Handles owner 'afterDelete' event and unlink all related records
     * @param \yii\base\Event $event event instance
     */
    public function afterDelete($event)
    {
        $this->owner->unlinkAll($this->relation, $this->deleteOnUnlink);
    }
```

https://github.com/iworker/ar-linkmany/commit/5ffa46aa3098ecffe65411459135e1b92d904baa

**Comment by klimov-paul** [comment]
Resolved by commit 103be867a1d5f112314f851b643afa5c236dc4ec
